model/roles.go

```diff
@@ -380,9 +381,61 @@ func (m *RoleManifest) GetRoleManifestDevPackageVersion(roles Roles, opinions *O
 
 // LookupRole will find the given role in the role manifest
 func (m *RoleManifest) LookupRole(roleName string) *Role {
+	if m.rolesByName == nil {
+		// This should only happen in tests where we manually build a manifest
+		m.rolesByName = make(map[string]*Role)
+		for _, role := range m.Roles {
+			m.rolesByName[role.Name] = role
+		}
+	}
 	return m.rolesByName[roleName]
 }
 
+// resolveLinks examines the BOSH links specified in the job specs and maps
+// them to the correct role / job that can be looked up at runtime
+func (m *RoleManifest) resolveLinks() validation.ErrorList {
+	errors := make(validation.ErrorList, 0)
+
+	// Resolve the consumers
+	for _, role := range m.Roles {
+		// Prefer finding providers in the same role
+		providingRoles := append(Roles{role}, m.Roles...)
+		for _, job := range role.Jobs {
+		iterateOverConsumers:
+			for idx, consumer := range job.LinkConsumes {
+				for _, providingRole := range providingRoles {
+					for _, providingJob := range providingRole.Jobs {
+						for _, provider := range providingJob.LinkProvides {
```

Comment:
Not enjoying the sextuply-nested loop, but this _does_ seem like cleaner code and probably doesn't matter…

Reply:
Definitely cleaner than the previous version. I like the simplicity of the logic.